src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPopupPanel.cs

```diff
@@ -121,24 +121,30 @@ private Rect CalculateFlyoutPlacement(Size desiredSize)
 				switch (placement)
 				{
 					case FlyoutPlacementMode.Top:
+						var possibleTopX = anchorRect.Left + halfAnchorWidth - halfChildWidth;
+						var possibleTopY = anchorRect.Top - PopupPlacementTargetMargin - desiredSize.Height;
 						finalPosition = new Point(
-							x: anchorRect.Left + halfAnchorWidth - halfChildWidth,
-							y: anchorRect.Top - PopupPlacementTargetMargin - desiredSize.Height);
+							x: possibleTopX > 0 ? possibleTopX : 0,
```

Comment:
```suggestion
							x: Math.Max(possibleTopX, 0),
```

Use `Math.Max()` here and similarly below, it's more readable.

Reply:
alright